src/test/java/org/zaproxy/zest/test/v1/ZestClientWindowResizeUnitTest.java

```diff
@@ -0,0 +1,131 @@
+        // Given / When
+        ZestClientWindowResize resize = new ZestClientWindowResize();
+        // Then
+        assertNull(resize.getWindowHandle());
+    }
+
+    @Test
+    void shouldBeEnabledByDefault() {
+        // Given / When
+        ZestClientWindowResize resize = new ZestClientWindowResize();
+        // Then
+        assertEquals(true, resize.isEnabled());
+    }
+
+    @Test
+    void shouldSetX() {
+        // Given
+        int x = 10;
+        ZestClientWindowResize resize = new ZestClientWindowResize();
+        // When
+        resize.setX(x);
+        // Then
+        assertEquals(x, resize.getX());
+    }
+
+    @Test
+    void shouldSetY() {
+        // Given
+        int y = 20;
+        ZestClientWindowResize resize = new ZestClientWindowResize();
+        // When
+        resize.setY(y);
+        // Then
+        assertEquals(y, resize.getY());
+    }
+
+    @Test
+    void shouldSerialiseAndDeserialise() {
+        // Given
+        ZestClientWindowResize original = new ZestClientWindowResize("windowHandle", 100, 500);
+        original.setEnabled(false);
+        // When
+        String serialisation = ZestJSON.toString(original);
+        ZestClientWindowResize deserialised =
+                (ZestClientWindowResize) ZestJSON.fromString(serialisation);
+        // Then
+        assertEquals(deserialised.getElementType(), original.getElementType());
+        assertEquals(deserialised.getWindowHandle(), original.getWindowHandle());
+        assertEquals(deserialised.getX(), original.getX());
+        assertEquals(deserialised.getY(), original.getY());
+        assertEquals(deserialised.isEnabled(), original.isEnabled());
+    }
+
+    @Test
+    void shouldDeepCopy() {
+        // Given
+        ZestClientWindowResize original = new ZestClientWindowResize("windowHandle", 100, 500);
+        original.setEnabled(false);
+        // When
+        ZestClientWindowResize copy = original.deepCopy();
+        // Then
+        assertThat(copy).isNotSameAs(original);
+        assertEquals(copy.getElementType(), original.getElementType());
+        assertEquals(copy.getWindowHandle(), original.getWindowHandle());
+        assertEquals(copy.getX(), original.getX());
+        assertEquals(copy.getY(), original.getY());
+        assertEquals(copy.isEnabled(), original.isEnabled());
+    }
+
+    @Test
+    void shouldFailToResizeIfWindowHandleNotFound() throws Exception {
+        // Given
+        ZestClientWindowResize resize = new ZestClientWindowResize("NoWindowHandle", 100, 500);
+        // When / Then
+        assertThrows(ZestClientFailException.class, () -> resize.invoke(testRuntime()));
+    }
+
+    @Test
+    void shouldResizeWindow() throws Exception {
+        // Given
+        ZestScript script = new ZestScript();
+        ZestBasicRunner runner = new ZestBasicRunner();
+        // When
+        script.add(new ZestClientLaunch("windowHandle", "firefox", "http://www.example.com"));
```

Comment:
This should use a local server, like other tests.

Reply:
okay